frappe/public/js/frappe/form/footer/timeline.js

```diff
@@ -128,7 +128,33 @@ frappe.ui.form.Timeline = Class.extend({
 				var name = $(this).parents(".timeline-item:first").attr("data-name");
 				me.delete_comment(name);
 				return false;
-			});
+			})
+            .on('click', '.edit', function() {
+                var is_editing = 'is-editing';
```

Comment:
When on keydown enter, the comment is submitted, then shift+enter needs to be used to make a new line(But that's the case in many editors anyway). Maybe change the edit icon to a "tick" so it gets more obvious that you should click on it once editing is finished?